### ISSUE TITLE
Add equal scaling for x y axis

### ISSUE THREAD

**Issue by gberrante** [swharden/ScottPlot#272]
Hi Scott,
thank you for this amazing work. 

I would like to suggest to add a feature that enables the user to have the same scaling for both x y axis. 

I tried to implement that by adjusting the axes limit from min(x,y) to max(x,y) and then using the aspect ratio of the plot and zoom out by the ratio the y axes. It works somehow but is not really a robust solution, the axis should be adjusted at any plot component resize. Another drawback that i did not solved yet is to re-center the plotted line in my case.

```c#
//equal axis scaling
double xmin = xarray.Min();
double xmax = xarray.Max();
double ymin = yarray.Min();
double ymax = yarray.Max();

double minimum = Math.Min(Math.Min(xmin,ymin),Math.Min(xmax,ymax));
double maximum = Math.Max(Math.Max(xmin, ymin), Math.Max(xmax, ymax));
this.axesXY.plt.Axis(minimum, maximum, minimum, maximum);
double Width = this.axesXY.plt.CoordinateToPixel(maximum, minimum).X- 
this.axesXY.plt.CoordinateToPixel(minimum,minimum).X;

double Height = this.axesXY.plt.CoordinateToPixel(minimum, maximum).Y - this.axesXY.plt.CoordinateToPixel(minimum, minimum).Y;

double ratio = Math.Abs(Width / Height);
this.axesXY.plt.AxisZoom(1,ratio);
this.axesXY.Render();
```

**Comment by swharden** [comment]
Hey @gberrante 

I might be misunderstanding your problem, but have you tried this?
```cs
// set x and y to the same axis range
plt.Axis(-10, 10, -10, 10);

// zoom in by the same fraction on both axes
plt.AxisZoom(2, 2);
```

**Comment by swharden** [comment]
oh, I think I understand now. You're using a user control. When you resize the user control, you want it to not distort the axes with respect to each other.  You want it to lock the *ratio* of horizontal and vertical spans together while resizing the user control. 

In other words: if you plot data shaped like a square, you want it to appear like a square (not a rectangle) after resizing the control.

Did I properly describe your goal?

**Comment by gberrante** [comment]
yes correct. for example i'm plotting a geometric shape and  i need that both axes are on the same scaling regardless the plot dimensions and aspect ratio

**Comment by swharden** [comment]
Pictures always help. Here's this issue described with Python

## problem

```python
import matplotlib.pyplot as plt
xs = [1, 1, 2, 2, 1]
ys = [1, 2, 2, 1, 1]
plt.plot(xs, ys, '-o')
plt.show()
```

![image](https://user-images.githubusercontent.com/4165489/75287006-9b224e00-57e7-11ea-89c0-b20e0e61662a.png)


## solution: scaled

```python
import matplotlib.pyplot as plt
xs = [1, 1, 2, 2, 1]
ys = [1, 2, 2, 1, 1]
plt.plot(xs, ys, '-o')
plt.axis('scaled')
plt.show()
```

![image](https://user-images.githubusercontent.com/4165489/75287054-b1300e80-57e7-11ea-8623-2d3b3c8f06b7.png)

## better solution: equal
```python
import matplotlib.pyplot as plt
xs = [1, 1, 2, 2, 1]
ys = [1, 2, 2, 1, 1]
plt.plot(xs, ys, '-o')
plt.axis('equal')
plt.show()
```

![image](https://user-images.githubusercontent.com/4165489/75287161-e3417080-57e7-11ea-97fe-f963916d9f44.png)

**Comment by swharden** [comment]
okay @gberrante I think I got this working!

I added two new functions: to `Plot.AxisScale()` and `Plot.AxisEqual()` but you don't need to call them.

To configure the FormsPlot control to do what you want, call `formsPlot.Configure(equalAxes: true)`

I'm working on a lot of parallel changes right now so I don't think I'll release this on NuGet as fast as I typically do (perhaps by the weekend), but it's ready to go if you download from the GitHub source.

![square](https://user-images.githubusercontent.com/4165489/75309347-cffdc780-581e-11ea-8cd4-54d41d370459.gif)

Feel free to open this issue again if it doesn't perform as expected!
